src/averageAges.js

```diff
@@ -14,14 +14,18 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+function calculateMenAverageAge(people, century = 0) {
+  const arrayMen = people.filter(
+    men => men.sex === 'm'
+    && (Math.ceil(men.died / 100) === century || century === 0));
+
+  const arrayLifesDurationMen = arrayMen.map(years => years.died - years.born);
 
+  const result = arrayLifesDurationMen.reduce(
+    (previousValue, currentValue) => previousValue + currentValue);
+
+  return result / arrayLifesDurationMen.length;
```

Comment:
you can mace function to calculate all these repeated code lines 24-27 (hint: write function before all your fubctions)